data_describe/anomaly_detection/detection.py

```diff
@@ -0,0 +1,548 @@
+        date_col (str, optional): Datetime column if data is timeseries. Defaults to None. If data is timeseries, date_col must be specified.
+        method (str, optional): Select method from this list. Only "arima" is supported.
+        estimator (optional): Fitted or instantiated estimator with a .predict() and .fit() method. Defaults to None.
+            If estimator is instantiated but not fitted, data_split must be specified.
+        n_periods (int, optional): Number of periods for timeseries anomaly detection. Default is None.
+        data_split (int, optional): Index to split the data into a train set and a test set. Defaults to None.
+            data_split must be specified if estimator is instantiated but not fitted.
+        **kwargs: Keyword arguments.
+
+    Raises:
+        ValueError: If method is not implemented.
+
+    Returns:
+        AnomalyDetectionWidget
+
+    """
+    # Timeseries indicator
+    if date_col:
+
+        # Supervised learning indicator
+        if target:
+            train, test = (
+                data[target][0:data_split],
+                data[target][data_split:],
+            )
+
+            # Default to ARIMA model
+            if not estimator:
+                from pmdarima.arima import auto_arima
+
+                estimator = auto_arima(train, random_state=1, **kwargs)
+
+            # make one-step forecast
+            predictions_df = stepwise_fit_and_predict(
+                train=train, test=test, n_periods=n_periods, estimator=estimator
+            )
+
+            # Post-processing for errors and confidence interval
+            predictions_df = _pandas_compute_anomalies_stats(
+                predictions_df, n_periods=n_periods
+            )
+
+        # Indicator for unsupervised learning
+        else:
+            raise ValueError(
+                "Unsupervised timeseries methods for anomaly detection are not yet supported. Please specify the target argument to continue."
+            )
+
+    # Indicator for regression and classification learning
+    else:
+        raise ValueError(
+            "Regression and Classification methods for anomaly detection are not yet supported. To specify timeseries methods, please specify the date_col."
+        )
+
+    return AnomalyDetectionWidget(
+        estimator=estimator,
+        method=method,
+        viz_data=predictions_df,
+        data_split=data_split,
+    )
+
+
+def stepwise_fit_and_predict(train, test, n_periods, estimator):
+    """Perform stepwise fit and predict for timeseries data.
+
+    Args:
+        train (Series): The training data.
+        test (Series): The testing data.
+        n_periods (int): The number of periods.
+        estimator: The estimator.
+
+    Returns:
+        predictions_df: DataFrame containing the ground truth, predictions, and indexed by the datetime.
+    """
+    logging.warning(
```

Comment:
I think this is more `info` level than `warning` since there isn't something unexpected or unusual happening here. Ideally we'd be able to integrate tqdm with a progress bar instead of logging messages.